src/util/server.ml

```diff
@@ -4,7 +4,10 @@ open Jsonrpc
 exception Failure of Response.Error.Code.t * string
 
 type t = {
-  file: Cil.file;
+  mutable file: Cil.file;
+  mutable version_map: (CompareCIL.global_identifier, Cil.global) Hashtbl.t;
+  mutable max_ids: VersionLookup.max_ids;
+  preprocess_and_merge: unit -> Cil.file;
   do_analyze: Analyses.increment_data -> Cil.file -> unit;
```

Comment:
I might've forgotten to bring this up in #522, but what's the reason for collecting these things into a record for server mode?

Reply:
The functions are there because I cannot access anything defined in `maingoblint.ml` without causing a cyclic dependency. The CIL stuff is there because `do_analyze` needs it.

I can also create a new module and move `do_analyze` & `preprocess_and_merge` to this new module to break the cyclic dependency if that is preferable.

Reply:
Moving these out of `Maingoblint` would be preferable indeed.

Reply:
Instead of moving these out of `Maingoblint`, I moved `main` into `Goblint`. Hope that's fine.

Reply:
I think that's kind of backwards now, `Maingoblint` should be the one to contain the main function for the goblint binary and the rest could go into `Goblint`. It should be equally easy to have them swapped because nothing else could depend on `Maingoblint` before anyway.

Reply:
OK, I swapped the two modules. Unfortunately, dune doesn't seem to like it when the executable name doesn't match the module name. So, I think I have to rename `src/goblint.exe` to `src/maingoblint.exe`.

Reply:
Ugh, ok. I thought `Maingoblint` was the module for `goblint.exe` since we got rid of the stupid `Goblint` module generation in #455, but I forgot that it still left the tiny module to do the `at_exit` hack.

I'd rather not have the binary itself renamed, so I guess we should just revert the module swap and live with the confusing naming for the time being (until `Maingoblint` gets split up even more to more descriptive modules). Sorry for the confusion!